bootstrap/internal/controllers/rke2config_controller.go

```diff
@@ -393,7 +462,23 @@ func (r *RKE2ConfigReconciler) joinControlplane(ctx context.Context, scope *Scop
 		Permissions: filePermissions,
 	}
 
-	// TODO: Implement adding additional files through API, coming in future PR
+	files, err := r.generateFileListIncludingRegistries(scope, ctx, configFiles)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	manifestFiles, err := generateFilesFromManifestConfig(ctx, r.Client, scope.ControlPlane.Spec.ManifestsConfigMapReference)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			scope.Logger.Error(err, "ConfigMap referenced by manifestsConfigMapReference not found!")
+			return ctrl.Result{}, err
+		}
+		scope.Logger.Error(err, "Problem when getting ConfigMap referenced by manifestsConfigMapReference")
+		return ctrl.Result{}, err
+	}
+
+	files = append(files, manifestFiles...)
+	// TODO: Implement adding additional files through API
```

Comment:
Do we need an issue for this so its not forgotten?

Reply:
This is actually already done in the `generateFileListIncludingRegistries` which adds the files from `RKE2ConfigSpec.Files`. So, I will remove the comment!